scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/Pipeline.kt

```diff
@@ -152,8 +153,11 @@ class Pipeline(
         ): Pair<Pipeline?, PipelineStatus.Error?> {
             val (topology, numSteps) = buildTopology(metadata, steps, kafkaDomainParams)
             val pipelineProperties = localiseKafkaProperties(kafkaProperties, metadata, numSteps, kafkaConsumerGroupIdPrefix, namespace)
+            pipelineProperties[STATE_DIR_CONFIG] = "/tmp/kafka-streams/${metadata.id}"
```

Comment:
Why was this change required? In `localiseKafkaProperties` we call `.withAppId`, which already passes a hash of metadata.id to the application ID, alongside other parts. By default, the state dir is `/tmp/kafka-streams/<app-id>`.

Reply:
Solved by modifying the Dockerfile